torch/nn/parallel/distributed.py

```diff
@@ -1190,6 +1284,9 @@ def forward(self, *inputs, **kwargs):
             output = _tree_unflatten_with_rref(
                 output_placeholders, treespec, output_is_rref
             )
+
+        # At the end of the forward pass, reset the grad buffer and grad views
+        self._clear_grad_buffer()
```

Comment:
this can be removed?

Reply:
We need to manually set grad buffer of these delayed all reduce to zero, and rebuild grad views if optimizer.zero_grad sets them to None.